docs/user-guide/data_types.rst

```diff
@@ -0,0 +1,183 @@
+Data types
```

Comment:
This file is a super useful read. I'm wondering what to do with it though. Were you thinking it would go under the `Advanced Topics` section in the user guide?

Reply:
No strong opinion from me. IMO our docs right now are not the most logically organized, so I anticipate some churn there in any case.